EpiAware/src/models.jl

```diff
@@ -1,39 +1,8 @@
-"""
-    log_infections(y_t, epimodel::EpiModel, latent_process;
-                   latent_process_priors,
-                   transform_function = exp,
-                   n_generate_ahead = 0,
-                   pos_shift = 1e-6,
-                   neg_bin_cluster_factor = missing,
-                   neg_bin_cluster_factor_prior = Gamma(3, 0.05 / 3))
-
-A Turing model for Log-infections undelying observed epidemiological data.
-
-This function defines a log-infections model for epidemiological data.
-It takes the observed data `y_t`, an `EpiModel` object `epimodel`, and a `latent_process`
-model. It also accepts optional arguments for the `latent_process_priors`, `transform_function`,
-`n_generate_ahead`, `pos_shift`, `neg_bin_cluster_factor`, and `neg_bin_cluster_factor_prior`.
-
-## Arguments
-- `y_t`: Observed data.
-- `epimodel`: Epidemiological model.
-- `latent_process`: Latent process model.
-- `latent_process_priors`: Priors for the latent process model.
-- `transform_function`: Function to transform the latent process into infections. Default is `exp`.
-- `n_generate_ahead`: Number of time steps to generate ahead. Default is `0`.
-- `pos_shift`: Positive shift to avoid zero values. Default is `1e-6`.
-- `neg_bin_cluster_factor`: Missing value for the negative binomial cluster factor. Default is `missing`.
-- `neg_bin_cluster_factor_prior`: Prior distribution for the negative binomial cluster factor. Default is `Gamma(3, 0.05 / 3)`.
-
-## Returns
-A named tuple containing the generated quantities `I_t` and `latent_process_parameters`.
-"""
-@model function log_infections(
+@model function make_epi_inference_model(
     y_t,
-    epimodel::EpiModel,
+    epimodel::AbstractEpiModel,
```

Comment:
Could you just briefly explain why we are handling the renewal etc differently to the underling latent process? I.e. why is one a function and one a AbstractEpiModel struct?

Reply:
I guess I am wondering if the priors for the latent process should be part of the latent process object in particular? Or if that is a pattern that doesn't make a lot of sense?

Reply:
And a similar comment for the observation model.

Reply:
> I guess I am wondering if the priors for the latent process should be part of the latent process object in particular? Or if that is a pattern that doesn't make a lot of sense?

Open to suggestions here!

So the mental model of `latent_process` is its a function that makes objects (in this case Turing models) rather than an object itself. So my first pass was to include some priors to give to it that says "Make Turing model with these priors". This interfaces nicely with `@submodel` to say "actually don't make that Turing model, but add it to the one you're declared inside of".  

Of course we could wrap these together so there is a self-contained object/struct that has "here is how we build latent processes" field and "these are the priors" field.